### DISCUSSION TITLE
How do pdfViewer control for slint in Liunx or Windows(10)?

### DISCUSSION THREAD

**Discussion by joyoes** [slint-ui/slint#3014] [General]
How do pdfViewer control for slint in Liunx or Windows(10)?
怎么使用slint在线浏览pdf file?

**Comment by tronical** [comment]
At the moment we don't have support for displaying PDFs built into Slint. What you could do is use something like Pdfium to render pages into images, put them into a `slint::Image`, and display them in a `Flickable`.

**Reply by joyoes** [reply]
Can not web chrome control for displaying PDFs into slint?

**Reply by tronical** [reply]
I don't know if there's a way to embed a web chrome control. Do you know?

**Reply by joyoes** [reply]
Can webkit control into slint?

**Reply by ogoffart** [reply]
The way it works is that you'd use the `Image { } ` element,  and you'd use the [pdfium-render](https://docs.rs/pdfium-render/latest/pdfium_render/) crate to generate an image for it.

We have a bunch of examples that use an Image element in such a way;

- https://github.com/slint-ui/slint/tree/master/examples/plotter uses an Image to draw a graph drawn with the plotter crate.  I believe this may be similar to what you could do with pdfium_render.  ([online demo](https://slint.dev/releases/1.1.0/demos/plotter/))
- https://github.com/slint-ui/slint/tree/master/examples/ffmpeg uses an Image to display a video using ffmpeg
- https://github.com/slint-ui/slint/tree/master/examples/opengl_texture embed a OpenGL texture in an Image
